### PR TITLE
refactor! [BREAKING] Switching to Project paradigm

### PR DESCRIPTION
**Pull request by JigarJoshi**
- One project contains one DB and it will extend with other offerings. 
- DB is replaced by Project in API.
- Event streaming APIs are removed for redesign.

### CODE REVIEW

File: server/v1/api.proto [hunk 1]
```diff
@@ -442,39 +442,39 @@ message Page {
...
+message CreateProjectResponse {
   // A detailed response message.
   string message = 1;
   // An enum with value set as "created".
   string status = 2;
 }
 
-message DropDatabaseRequest {
-  // Drop database with this name. **Note**: Deletes all the collections in the database.
+message DropProjectRequest {
```
Comment (himank) [inline server/v1/api.proto:460]: `DeleteProjectRequest`

File: server/v1/api.proto [hunk 2]
```diff
@@ -442,39 +442,39 @@ message Page {
...
   // Use with caution.
-  string db = 1;
-  // Optionally specify a database branch name to perform operation on
+  string project = 1;
+  // Optionally specify a project branch name to perform operation on
   string branch = 3;
 }
 
-message DropDatabaseResponse {
+message DropProjectResponse {
```
Comment (himank) [inline server/v1/api.proto:468]: ^

File: server/v1/api.proto [hunk 3]
```diff
@@ -884,87 +884,87 @@ service Tigris {
...
-      tags: "Databases"
+      summary: "Create Project"
+      tags: "Projects"
     };
   }
 
-  // Drop database deletes all the collections in this database along with all of the documents, and associated metadata for these collections.
-  rpc DropDatabase(DropDatabaseRequest) returns (DropDatabaseResponse) {
+  // Drop project deletes all the collections in this project along with all of the documents, and associated metadata for these collections.
+  rpc DropProject(DropProjectRequest) returns (DropProjectResponse) {
```
Comment (himank) [inline server/v1/api.proto:940]: `DeleteProject`

File: server/v1/api.proto [hunk 4]
```diff
@@ -884,87 +884,87 @@ service Tigris {
...
     option (google.api.http) = {
-      delete : "/v1/databases/{db}/drop"
+      delete : "/v1/projects/{project}/drop"
```
Comment (himank) [inline server/v1/api.proto:942]: `"/v1/projects/{project}/delete"`

Comment (JigarJoshi) [reply server/v1/api.proto:874]: fixed.

File: server/v1/api.proto (hunk 3 above)
Comment (JigarJoshi) [reply server/v1/api.proto]: fixed.

File: server/v1/api.proto (hunk 1 above)
Comment (JigarJoshi) [reply server/v1/api.proto]: fixed.

**Review by himank — APPROVED** [review]

**Pull request merged**